extensions/wikia/Chat2/ChatBanListSpecial_helper.php

```diff
@@ -0,0 +1,239 @@
+	}
+
+	function getUserName() { return $this->userName; }
+
+	function getLimit() { return $this->limit; }
+
+	function getOffset() { return $this->offset; }
+
+	function getOrder() { return $this->order; }
+
+	public function loadData() {
+
+		wfProfileIn( __METHOD__ );
+
+		/* initial values for result */
+		$data = [
+			'cnt'      => 0,
+			'sColumns' => 'timestamp,target,expires,blockedBy,reason',
+			'data'     => [ ],
+		];
+
+		/* initial conditions for SQL query */
+		$where = [
+			'cbu_wiki_id' => $this->cityId,
+			"cbu_user_id != ''",
+			"end_date > '" . wfTimestamp( TS_MW ) . "'",
+		];
+
+		/* filter: user name */
+		$userId = User::IdFromName( $this->userName );
+
+		if ( $userId !== null ) {
+			$where[] = " cbu_user_id = " . $this->db->addQuotes( $userId );
+		}
+
+		/* number of records */
+		$row = $this->db->selectRow(
+			$this->table,
+			[ 'count(0) as cnt' ],
+			$where,
+			__METHOD__
+		);
+		if ( is_object( $row ) ) {
+			$data['cnt'] = $row->cnt;
+		}
+
+		if ( $data['cnt'] > 0 ) {
+			/* select records */
+			$oRes = $this->db->select(
+				[
+					$this->table,
+				],
+				[
+					'start_date',
+					'cbu_user_id',
+					'end_date',
+					'cbu_admin_user_id',
+					'reason',
+				],
+				$where,
+				__METHOD__,
+				[
+					'ORDER BY' => $this->order,
+					'LIMIT'    => $this->limit,
+					'OFFSET'   => intval( $this->offset ),
+				]
+			);
+
+			$data['data'] = [ ];
+			while ( $row = $this->db->fetchObject( $oRes ) ) {
+
+				$user = User::newFromId( $row->cbu_user_id );
+				$admin = User::newFromId( $row->cbu_admin_user_id );
+
+				$data['data'][] = [
+					'timestamp'    => $this->wg->Lang->timeanddate( $row->start_date, true ),
+					'user'         => Linker::link( $user->getUserPage(), $user->getName() ),
+					'user_actions' => $this->getUserLinks( $user ),
+					'expires'      => $this->wg->Lang->formatExpiry( $row->end_date, true ),
+					'admin_user'   => Linker::link( $admin->getUserPage(), $admin->getName() ),
+					'admin_links'  => $this->getUserLinks( $admin ),
+					'reason'       => Linker::commentBlock( $row->reason ),
+				];
+
+			}
+			$this->db->freeResult( $oRes );
+
+		}
+
+		wfProfileOut( __METHOD__ );
+
+		return $data;
+	}
+
+	private function getUserLinks( $user ) {
+
+		$userIsBlocked = $this->wg->User->isBlocked( true, false );
+		$oEncUserName = urlencode( $user->getName() );
+		$links = [
+			0 => "",
+			1 => Linker::link(
+				Title::newFromText( 'Contributions', NS_SPECIAL ),
+				$this->wg->Lang->ucfirst( wfMsg( 'contribslink' ) ),
+				[ 'target' => $oEncUserName]
+			),
+		];
+
+		if ( !empty( $this->wg->EnableWallExt ) ) {
+			$oUTitle = Title::newFromText( $user->getName(), NS_USER_WALL );
+			$msg = 'wall-message-wall-shorten';
+		} else {
+			$oUTitle = Title::newFromText( $user->getName(), NS_USER_TALK );
+			$msg = 'talkpagelinktext';
+		}
+
+		if ( $oUTitle instanceof Title ) {
+			$links[0] = Linker::link( $oUTitle, $this->wg->Lang->ucfirst( wfMsg( $msg ) ) );
```

Comment:
We will have to use per-language caching here (surrogate key should be changed for that), otherwise this code will suffer from the same i18n caching problems as ListUsers.